universal-application-tool-0.0.1/app/views/components/QuestionBank.java

```diff
@@ -140,4 +149,44 @@ private ContainerTag renderQuestionDefinition(QuestionDefinition definition) {
                 addButton);
     return questionDiv.with(icon, content);
   }
+
+  /**
+   * Used to filter questions in the question bank.
+   *
+   * <p>Questions that are filtered out:
+   *
+   * <ul>
+   *   <li>If there is at least one question in the current block, all repeater questions
+   *   <li>If this is a repeated block only show the appropriate repeated questions
+   *   <li>Questions already in the program
+   * </ul>
+   */
+  private ImmutableList<QuestionDefinition> filterQuestions() {
+    // A repeater block cannot add any more questions
+    if (blockDefinition.isRepeater()) {
```

Comment:
Suggestion (non-blocking): it seems like this would be the better place to check if the block has a repeater question yet or not

Reply:
I don't understand the comment. Isn't this the place that is checking if the block has a repeater question?

Reply:
If you're not using `repeaterQuestionId` this can be ignored!